File: SFBackend/src/main/java/com/sfinance/SFBackend/Entity/Product.java
package com.sfinance.SFBackend.Entity;

import lombok.*;

import javax.persistence.*;

@Entity
@Getter
@Setter
@ToString
@AllArgsConstructor
@NoArgsConstructor

public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(nullable = false, updatable = false)
    private Long id;
    private String productName;
    private Double boughtPrice;
    private Double soldPrice;
    private String nameBrand;
    private String nameCategory;
    private Integer quantity;
}

File: SFFrontend/src/app/model/product.ts
export class Product{

    id: number;
    productName: string;
    boughtPrice: string;
    soldPrice: string;
    nameBrand: string;
    nameCategory: string;
    quantity: string;

    constructor(){
        this.productName = '';
        this.boughtPrice = '';
        this.soldPrice = '';
        this.nameBrand = '';
        this.nameCategory = '';
        this.quantity = '';
    }
}